tests/e2e/svmotion_detached_volume.go

```diff
@@ -163,4 +169,268 @@ var _ = ginkgo.Describe("[csi-block-vanilla] [csi-block-vanilla-parallelized] Re
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfs2URL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfs2URL)
+		}()
+
+		policyID, policyName := createTagBasedPolicy(
+			ctx, pc, map[string]string{categoryName: tagName})
+		defer func() {
+			deleteStoragePolicy(ctx, pc, policyID)
+		}()
+		scParameters[scParamStoragePolicyName] = policyName
+		storageclass, pvclaim, err := createPVCAndStorageClass(client,
+			namespace, nil, scParameters, "", nil, "", false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		pvcs = append(pvcs, pvclaim)
+
+		defer func() {
+			ginkgo.By("Delete the SCs created")
+			err := client.StorageV1().StorageClasses().Delete(ctx, storageclass.Name, *metav1.NewDeleteOptions(0))
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Deleted the SPBM polices created")
+			_, err = pc.DeleteProfile(ctx, []pbmtypes.PbmProfileId{*policyID})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Verify the PVCs created in step 3 are bound")
+		pvs, err := fpv.WaitForPVClaimBoundPhase(client, pvcs, framework.ClaimProvisionTimeout)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		volumeID := pvs[0].Spec.CSI.VolumeHandle
+
+		defer func() {
+			err := fpv.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			err = e2eVSphere.waitForCNSVolumeToBeDeleted(pvs[0].Spec.CSI.VolumeHandle)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Verify that the created CNS volumes are compliant and have correct policy id")
+		storagePolicyExists, err := e2eVSphere.VerifySpbmPolicyOfVolume(volumeID, policyName)
+		e2eVSphere.verifyVolumeCompliance(volumeID, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(storagePolicyExists).To(gomega.BeTrue(), "storage policy verification failed")
+
+		ginkgo.By("Creating a pod")
+		pod, err := createPod(client, namespace, nil, pvcs, false, "")
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		filePath := "/mnt/volume1/file1.txt"
+		filePath2 := "/mnt/volume1/file2.txt"
+
+		//Write data on file.txt on Pod
+		data := "This file is written by Pod"
+		ginkgo.By("write to a file in pod")
+		writeDataOnFileFromPod(namespace, pod.Name, filePath, data)
+
+		defer func() {
+			ginkgo.By("Delete the pod created")
+			err := fpod.DeletePodWithWait(client, pod)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		ginkgo.By("Verify if VolumeID is created on the given datastores")
+		dsUrlWhereVolumeIsPresent := fetchDsUrl4CnsVol(e2eVSphere, volumeID)
+		framework.Logf("Volume is present on %s", dsUrlWhereVolumeIsPresent)
+		e2eVSphere.verifyDatastoreMatch(volumeID, datastoreUrls)
```

Comment:
We should add Error handling here

Reply:
the error handling for datastore match is already present inside the method verifyDatastoreMatch